h2o-py/h2o/explanation/_explain.py

```diff
@@ -1617,56 +1688,40 @@ def model_correlation_heatmap(
         ]
     else:
         top_n = len(models)
-
     is_classification = frame[models[0].actual_params["response_column"]].isfactor()[0]
-
     models = models[:min(len(models), top_n)]
-    predictions = np.empty((len(models), frame.nrow),
-                           dtype=np.object if is_classification else np.float)
+    predictions = []
     with no_progress():
         for idx, model in enumerate(models):
-            predictions[idx, :] = np.array(model.predict(frame)["predict"]
-                                           .as_data_frame(use_pandas=False, header=False)) \
-                .reshape(frame.nrow)
+            predictions.append(model.predict(frame)["predict"])
 
     if is_classification:
         corr = np.zeros((len(models), len(models)))
         for i in range(len(models)):
             for j in range(len(models)):
                 if i <= j:
-                    corr[i, j] = (predictions[i, :] == predictions[j, :]).mean()
+                    corr[i, j] = (predictions[i] == predictions[j]).mean()[0]
                     corr[j, i] = corr[i, j]
     else:
-        corr = np.corrcoef(predictions)
-
+        try:
+            from io import StringIO
+        except ImportError:
+            from StringIO import StringIO
```

Comment:
this is standard lib, the import could go on the top of the file: looks a bit noisy here.

Reply:
btw the order is reverted: Py 2.7 has both `io.StringIO` (for unicode only) and `StringIO.StringIO` (8 bits strings or unicode).
So you for max compatibility you need to use:
```python
try:
    from StringIO import StringIO  # py2 (first as py2 also has io.StringIO, but only with unicode support)
except:
    from io import StringIO  # py3
```
Also, when doing compatibility imports, please write a short comment for each import telling when each is expected to work.

Reply:
Thank you! I didn't know this.